crates/anvil/src/eth/miner.rs

```diff
@@ -25,12 +25,24 @@ pub struct Miner {
     ///
     /// This will register the task so we can manually wake it up if the mining mode was changed
     inner: Arc<MinerInner>,
+    /// Transactions included into the pool before any others are.
+    /// Done once on startup.
+    force_transactions: Option<Vec<Arc<PoolTransaction>>>,
 }
 
 impl Miner {
-    /// Returns a new miner with that operates in the given `mode`
-    pub fn new(mode: MiningMode) -> Self {
-        Self { mode: Arc::new(RwLock::new(mode)), inner: Default::default() }
+    /// Returns a new miner with that operates in the given `mode`.
+    /// Specifying some force transactions will cause a block to be mined with those transactions
+    /// as soon as the miner is polled.
+    pub fn new(mode: MiningMode, force_transactions: Vec<PoolTransaction>) -> Self {
```

Comment:
I'd like to keep new as before and instead have a setter for force_transactions:

```rust
fn with_forced_transactions(self, Option<tx>s) -> Self
```

Reply:
updated